src/test/java/io/github/perrymant/moneymaker/ApplicationTest.java

```diff
@@ -44,21 +63,63 @@
             "        ╚════════════╧══════════════════╧════════╧═════════╧═════════════╝\n";
 
     @Test
-    public void printsHelpMessageIfHelpIsGiven() {
+    public void noArgumentsIsGiven_printsError() {
+        target.start(new String[]{});
+        Assert.assertEquals(ERROR, logger.getMessage());
+    }
+
+    @Test
+    public void reportIsGiven_printsReport() {
+        target.start(new String[]{"report"});
+        Assert.assertEquals(REPORT, logger.getMessage());
+    }
+
+    @Test
+    public void reportWithOtherArgsIsGiven_printsReport() {
+        target.start(new String[]{REPORT_MORE_ARGS});
```

Comment:
This still only has one argument as there's only one element in the array. The fact that this is a string with multiple words is irrelevant.
This test is checking the equivalent of:
```
moneymaker "report more args given"
```
note the quotes.

Not sure this is worth testing right now.

Reply:
I Don't want to global comment this - but there is no option of a reply, so I will put my reply here!
- I ran Main with `report more args given` as Program Arguments in the Configuration for Main. `"report".equals(args[0])` in the Application class results in tacking multiple arguments given (as an array). 
- So I think this test is desirable as it deals with the logic that you pointed out (4 conversations above this one).

Reply:
Are you saying that we should test the case when a user calls `moneymaker "report more args given"`, and that we expect it to be treated as an invalid argument? If so, I'm cool with it and just mark this conversation as resolved.

Technical communication tip: "4 conversations above this one" is annoying to figure out (and not guaranteed to be accurate). It is possible to link to comments in GitHub. Left as an exercise for the reader 😉